schemas/webpack.d.ts

```diff
@@ -0,0 +1,360 @@
+        sourcePrefix?: string;
+
+        // If `output.libraryTarget` is set to umd and `output.library` is set, setting this to true will name the AMD module.
+        umdNamedDefine?: boolean;
+    }
+
+    interface OutputAuxiliaryComment {
+        amd?: string;
+        commonjs?: string;
+        commonjs2?: string;
+        root?: string;
+    }
+
+    interface OutputDevtoolLineToLine {
+        exclude?: string;
+        include?: string;
+        test?: string;
+    }
+
+    interface Resolve {
+        alias?: { key: string } | ResolveAlias[];
+        aliasFields?: [string | string[]];
+        cachePredicate?: Function;
+        descriptionFiles?: string[];
+        enforceExtension?: boolean;
+        enforceModuleExtension?: boolean;
+        extensions?: string[];
+        fileSystem?: any; // TODO missing type
+        mainFields?: [string | string[]];
+        mainFiles?: string[];
+        moduleExtensions?: string[];
+        modules?: string[];
+        plugins?: any[];
+        resolver?: any; // TODO missing type
+        symlinks?: boolean;
+        unsafeCache?: boolean | Object;
+    }
+
+    interface ResolveAlias {
+        alias?: string;
+        name?: string;
+        onlyModule?: boolean;
+    }
+
+    type EntryItem = string | string[];
+
+    type OutputLibraryTarget =
+        "var"
+        | "assign"
+        | "this"
+        | "window"
+        | "global"
+        | "commonjs"
+        | "commonjs2"
+        | "commonjs-module"
+        | "amd"
+        | "umd"
+        | "umd2"
+        | "jsonp";
+
+    type Stats =
+        "none"
+        | "errors-only"
+        | "minimal"
+        | "normal"
+        | "verbose";
+
+    type Target =
+        "web"
+        | "webworker"
+        | "node"
+        | "async-node"
+        | "node-webkit"
+        | "electron";
+
+    type Externals = string | {key: string} | ((context: any, request: any, callback: (err: any, result: any)=> any) => void) | RegExp;
+
+    type RuleSetCondition = RegExp | string | Function | RuleSetConditionObject;
+    type RuleSetConditionRecursive = RuleSetCondition | RuleSetCondition[];
+    type RuleSetConditions = RuleSetConditionRecursive[];
+    type RuleSetUse = RuleSetUseItem | RuleSetUseItem[];
+    type RuleSetUseItem = string | RuleSetUseItemObject;
+    type RuleSetQuery = Object | string;
+}
+
+interface Webpack {
+    /*
+     Set the value of `require.amd` and `define.amd`.
+     */
+    amd?: any; // TODO missing type
+
+    /*
+     Report the first error as a hard error instead of tolerating it.
+     */
+    bail?: boolean;
+
+    /*
+     Cache generated modules and chunks to improve performance for multiple incremental builds.
+
+     Boolean: You can pass `false` to disable it.
+     Object: You can pass an object to enable it and let webpack use the passed object as cache.This way you can share the cache object between multiple compiler calls.
+     */
+    cache?: (boolean | Object);
```

Comment:
Suggest to add an Object interface. Reason is the current definition will let `cache = "true"` pass, which should not.

```
interface Object {
[key: string]: any;
}
```

After adding this, the above statement will cause error.